### PR TITLE
fix(config): Merge auth and mailer configs

### LINKED ISSUE #1733: Use the same config for mailer bin and key_server bin
Stuff from https://github.com/mozilla/fxa-auth-server/blob/master/mailer/config.js

should go into 

https://github.com/mozilla/fxa-auth-server/blob/master/config/index.js

### PR DESCRIPTION
**Pull request by vbudhram**
From devops work week. Need to verify it doesn't break any other repos. DO NOT MERGE

Fixes #1733

### CODE REVIEW

File: bin/mailer_server.js [hunk 1]
```diff
@@ -3,10 +3,10 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 var restify = require('restify')
-var config = require('../mailer/config')
+var config = require('../config')
 
 var log = require('../lib/senders/log')('server')
-var mailConfig = config.get('mail')
+var mailConfig = config.get('smpt')
```
Comment (vladikoff) [inline bin/mailer_server.js]: smpt?

File: lib/senders/legacy_index.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = function (log, config, translator, sender) {
   return require('./templates')()
     .then(function (templates) {
       return {
-        email: new Mailer(translator, templates, config.mail, sender),
+        email: new Mailer(translator, templates, config.smtp, sender),
```
Comment (philbooth) [inline lib/senders/legacy_index.js]: Is there any reason for this module to exist, now that the configs are merged? Can we ditch this and switch everything over to just require `lib/senders`?

File: mailer/CHANGELOG.md [hunk 3]
```diff
@@ -1,229 +0,0 @@
-<a name="1.81.0"></a>
```
Comment (philbooth) [inline mailer/CHANGELOG.md]: In an ideal world, would we merge this with the other `CHANGELOG.md`? Or keep it separately somewhere? Obviously it's no big deal, but if it was worth generating I wonder if it's worth keeping is all.

File: bin/mailer_server.js [hunk 4]
```diff
@@ -81,10 +81,10 @@ P.all(
     )
 
     api.listen(
-      config.get('port'),
-      config.get('host'),
+      config.get('smtp.port'),
```
Comment (vladikoff) [inline bin/mailer_server.js]: can we just use `mailConfig` here? or do we need `mailConfig`? (see above)

File: mailer/CHANGELOG.md (hunk 3 above)
Comment (vladikoff) [inline mailer/CHANGELOG.md]: Let's keep to remember which features landed. We can move it to the `docs/` dir though, this way we can get rid of the `mailer` dir

Comment (vbudhram) [inline mailer/CHANGELOG.md]: Sounds good!

**Comment by vbudhram** [comment]
@vladikoff @philbooth Taking the WIP label off, r?

**Comment by philbooth** [comment]
Looks good to me @vbudhram, r+!

**Pull request merged**